src/order/filter/basic.lean

```diff
@@ -2588,19 +2588,29 @@ lemma tendsto.not_tendsto {f : α → β} {a : filter α} {b₁ b₂ : filter β
   ¬ tendsto f a b₂ :=
 λ hf', (tendsto_inf.2 ⟨hf, hf'⟩).ne_bot.ne hb.eq_bot
 
+lemma comap_ite (p : α → Prop) [decidable_pred p] (f g : α → β) (l : filter β) :
+  comap (λ x, ite (p x) (f x) (g x)) l = comap f l ⊓ 𝓟 {x | p x} ⊔ comap g l ⊓ 𝓟 {x | ¬p x} :=
```

Comment:
nit: is the following better since it preserves the property of having `p` before `f` and `g`?
```suggestion
  comap (λ x, ite (p x) (f x) (g x)) l = 𝓟 {x | p x} ⊓ comap f l ⊔ 𝓟 {x | ¬p x} ⊓ comap g l :=
```

Reply:
We have `principal` as the second argument of `inf` in the definition of `nhds_within` and most lemmas (try `git grep inf_principal`).